src/tests/ftest/recovery/ddb.py

```diff
@@ -0,0 +1,601 @@
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import os
+import re
+from ClusterShell.NodeSet import NodeSet
+
+from apricot import TestWithServers
+from general_utils import report_errors, run_pcmd, insert_objects, \
+    distribute_files, DaosTestError, get_random_string, copy_remote_to_local
+from ddb_utils import DdbCommand
+from exception_utils import CommandFailure
+
+
+class DdbTest(TestWithServers):
+    """Test ddb subcommands.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DdbTest object."""
+        super().__init__(*args, **kwargs)
+        # Generate random keys and data to insert into the object.
+        self.random_dkey = get_random_string(10)
+        self.random_akey = get_random_string(10)
+        self.random_data = get_random_string(10)
+
+    def get_vos_file_path(self):
+        """Get the VOS file path.
+
+        If there are multiple VOS files, returns the first file obtained by "ls".
+
+        Returns:
+            str: VOS file path such as /mnt/daos/<pool_uuid>/vos-0
+
+        """
+        hosts = NodeSet(self.hostlist_servers[0])
+        vos_path = os.path.join("/mnt/daos", self.pool.uuid.lower())
+        command = " ".join(["sudo", "ls", vos_path])
+        cmd_out = run_pcmd(hosts=hosts, command=command)
+
+        # return vos_file
+        for file in cmd_out[0]["stdout"]:
+            # Assume the VOS file has "vos" in the file name.
+            if "vos" in file:
+                self.log.info("vos_file: %s", file)
+                return file
+
+        self.fail("vos file wasn't found in /mnt/daos/{}".format(self.pool.uuid.lower()))
+
+        return None  # to appease pylint
+
+    def test_recovery_ddb_ls(self):
+        """Test ddb ls.
```

Comment:
You might want to add JIRA DAOS-10830 to all the def test_.. description

Reply:
I'll update in the subsequent PR. Thanks.